From 243eb0bb1775752cc1cfb03e75a90fbcc3969013 Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Mon, 5 Jun 2017 15:09:30 -0700
Subject: [PATCH 2/2] Use `dotnet` from `DOTNET_TOOL_DIR` if it is set

This allows the outer build to point at a specific version of `dotnet`
instead of just picking whatever is set on `$PATH`
---
 build/Build.targets | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/build/Build.targets b/build/Build.targets
index 0f34011..527718a 100644
--- a/build/Build.targets
+++ b/build/Build.targets
@@ -1,12 +1,17 @@
 <Project ToolsVersion="15.0">
 
+    <PropertyGroup>
+        <DotNetCommand Condition="'$(DOTNET_TOOL_DIR)' != ''">$(DOTNET_TOOL_DIR)/dotnet</DotNetCommand>
+        <DotNetCommand Condition="'$(DotNetCommand)' == ''">dotnet</DotNetCommand>
+    </PropertyGroup>
+
     <Target Name="Package"
             DependsOnTargets="RestoreSrcPackages">
 
-        <Exec Command='dotnet pack --no-build --output $(ArtifactsDir)/nupkgs /p:Version=$(FSharpSdkVersion)' 
+        <Exec Command='$(DotNetCommand) pack --no-build --output $(ArtifactsDir)/nupkgs /p:Version=$(FSharpSdkVersion)' 
               WorkingDirectory="$(RepoRoot)/src/FSharp.Sdk"/>
 
-        <Exec Command='dotnet pack --no-build --output $(ArtifactsDir)/nupkgs /p:Version=$(FSharpNETSdkVersion)' 
+        <Exec Command='$(DotNetCommand) pack --no-build --output $(ArtifactsDir)/nupkgs /p:Version=$(FSharpNETSdkVersion)' 
               WorkingDirectory="$(RepoRoot)/src/FSharp.NET.Sdk"/>
     </Target>
 
@@ -25,7 +30,7 @@
             Outputs="@(RestoreSrcPackagesInput->'%(RelativeDir)/obj/project.assets.json');@(RestoreSrcPackagesInput->'%(RelativeDir)/obj/%(Filename).csproj.nuget.g.props')">
         <!-- Restore test projects, if needed (with lock files up-to-date check) -->
 
-        <Exec Command='dotnet restore %(RestoreSrcPackagesInput.FullPath) --packages $(NuGetPackageRoot) --no-cache -v n' />
+        <Exec Command='$(DotNetCommand) restore %(RestoreSrcPackagesInput.FullPath) --packages $(NuGetPackageRoot) --no-cache -v n' />
     </Target>
 
-</Project>
\ No newline at end of file
+</Project>
-- 
1.8.3.1

